### DISCUSSION TITLE
Write EXIF DateTimeOriginal example, please

### DISCUSSION THREAD

**Discussion by GF-Huang** [SixLabors/ImageSharp#1235] [Q&A] [answered]
How to do that?

**Comment by JimBobSquarePants** [comment, marked as answer]
Hi @GF-Huang here's an example. 

``` c#
using (var image = new Image<Rgba32>(100, 100))
{
    // Create a new profile, since we crated the image.
    image.Metadata.ExifProfile = new ExifProfile();

    const string format = "yyyy:MM:dd HH:mm:ss";
    var now = DateTime.Now;
    var dt = now.ToString(format);

    // Set the metadata value.
    image.Metadata.ExifProfile.SetValue(ExifTag.DateTimeOriginal, dt);

    // Get the value.
    var dt2 = image.Metadata.ExifProfile.GetValue(ExifTag.DateTimeOriginal).Value;
    DateTime.TryParseExact(dt2, format, CultureInfo.InvariantCulture, DateTimeStyles.AssumeLocal, out DateTime now2);
}
```

https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Metadata.Profiles.Exif.ExifTag.html?q=DateTimeOriginal

@dlemstra I wonder.... date and times are always formatted in the same way yeah? Should we provide an abstraction that allows direct get/set of `DateTime` instances, having to parse is pretty ugly.

**Reply by dlemstra** [reply]
We could probably introduce an `ExifDateTime` `ExifValue` to handle this and do the conversion to and from string there. I do wonder what the timezone of that `DateTime` object should be though.

**Reply by JimBobSquarePants** [reply]
There's a TimeZoneOffset in 2.3.1 which we could use to offset DateTimeOriginal, ModifyDate and other local datetime tags. GPSDateStamp/GPSTimeStamp are stored in UTC so can be used (theoretically) [for deducing the original timezone](https://exiftool.org/forum/index.php?topic=1735.0)

**Reply by dlemstra** [reply]
But that would mean that the values would need to have knowledge about eachother and that would be strange to me. The value now has no knowledge of the profile it is in.

**Reply by JimBobSquarePants** [reply]
Yeah, you're right. that would be strange. We should just parse via `DateTimeStyles.AssumeLocal`. Any calculation can be  performed by the user.

**Comment by JimBobSquarePants** [comment]
Yo @GF-Huang did that answer your question?

**Reply by GF-Huang** [reply]
Sorry, I forget to read this issue..
